org.lflang/src/org/lflang/generator/c/CGenerator.xtend

```diff
@@ -3869,194 +3896,136 @@ class CGenerator extends GeneratorBase {
                 «structType»* «nameOfSelfStruct» = new_«reactorClass.name»();
             ''')
         }
-        
-        generateTraceTableEntries(instance, initializeTriggerObjects, federate)
-              
-        generateReactorInstanceExtension(initializeTriggerObjects, instance, federate)
 
         // Generate code to initialize the "self" struct in the
         // _lf_initialize_trigger_objects function.
         pr(initializeTriggerObjects, "//***** Start initializing " + fullName)
-
-        // Start with parameters.
-        generateParameterInitialization(initializeTriggerObjects, instance)
+        
+        generateTraceTableEntries(instance, instance.actions, instance.timers)
```

Comment:
There seems to be duplicate function calls here that are also in `generateFederate`. I wonder if these two could be consolidated, or the latter could be moved to `federated.CGeneratorExtension.java`

Reply:
Yes, this is on the to-do list. I've added a FIXME.